code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm

```diff
@@ -272,6 +272,13 @@ datum/reagent/water/reaction_obj(var/obj/O, var/volume)
 			cube.Expand()
 	return
 
+datum/reagent/water/reaction_obj(var/obj/O, var/volume)
+	src = null
+	if(istype(O,/obj/item/dehy_carp))
+		var/obj/item/dehy_carp/dehy = O
+		dehy.Swell() // Makes a carp
+		return
```

Comment:
There's gotta be a less intrusive way to do this. See how monkey cubes handle it, as this is effectively the same item except with a carp.